compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -338,5 +346,20 @@ class ReifyQuotes extends MacroTransformWithImplicits {
             checkLevel(mapOverTree(enteredSyms))
         }
       }
+
+    /** InlineSplice is used to detect cases where the expansion
+     *  consists of a (possibly multiple & nested) block or a sole expression.
+     */
+    object InlineSplice {
```

Comment:
I don't know how performance sensitive this code is but you could return a `Tree` instead of `Option[Select]` to avoid boxing (and `EmptyTree` instead of `None`)

Reply:
We need that `.get` method on `unapply`s return type so isn't `Option` unavoidable?

Reply:
I believe `Tree` qualifies for both [product pattern](http://dotty.epfl.ch/docs/reference/changed/pattern-matching.html#product-pattern) and [name based pattern](http://dotty.epfl.ch/docs/reference/changed/pattern-matching.html#name-based-pattern). And scalac supports name based pattern